diff --git a/drivers/clocksource/Kconfig b/drivers/clocksource/Kconfig
index 0f5e3983951a..b2b07aff1d6b 100644
--- a/drivers/clocksource/Kconfig
+++ b/drivers/clocksource/Kconfig
@@ -543,7 +543,7 @@ config CLKSRC_VERSATILE
 	bool "ARM Versatile (Express) reference platforms clock source" if COMPILE_TEST
 	depends on GENERIC_SCHED_CLOCK
 	select TIMER_OF
-	default y if (ARCH_VEXPRESS || ARCH_VERSATILE) && ARM
+	default y if (ARCH_VEXPRESS || ARCH_VERSATILE) && (ARM || ARM64)
 	help
 	  This option enables clock source based on free running
 	  counter available in the "System Registers" block of
