diff --git a/SConstruct b/SConstruct
index ef4d154312..25bb19e7bf 100755
--- a/SConstruct
+++ b/SConstruct
@@ -187,6 +187,7 @@ main = Environment(tools=[
 
 main.Tool(SCons.Tool.FindTool(['gcc', 'clang'], main))
 main.Tool(SCons.Tool.FindTool(['g++', 'clang++'], main))
+main.Tool('compilation_db')
 
 Export('main')
 
diff --git a/src/SConscript b/src/SConscript
index 52051b3256..f8e3d80945 100644
--- a/src/SConscript
+++ b/src/SConscript
@@ -64,6 +64,9 @@ if env['CONF']['USE_EFENCE']:
 # Children need to see the environment
 Export('env')
 
+# Generate compile_commands.json for language servers
+env.CompilationDatabase('compile_commands.json')
+
 build_env = list(env['CONF'].items())
 
 from code_formatter import code_formatter
